lib/semian/atomic_enum.rb

```diff
@@ -0,0 +1,38 @@
+require 'forwardable'
+
+module Semian
+  class AtomicEnum #:nodoc:
+    extend Forwardable
+
+    def_delegators :@integer, :destroy
+
+    def initialize(symbol_list)
+      @integer = Semian::AtomicInteger.new
+      initialize_lookup(symbol_list)
+    end
+
+    def increment(val = 1)
+      @integer.value = (@integer.value + val) % @sym_to_num.size
```

Comment:
Most are really small stubs, since when I put in the SharedMemoryObject stuff, Simon said that was already too big, so I had to rip that out, and so the mutex stuff, which I would put in there, is gone. Maybe a rename to just `Integer` than `AtomicInteger` ?